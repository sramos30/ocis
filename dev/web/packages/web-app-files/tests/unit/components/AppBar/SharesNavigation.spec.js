import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import VueRouter from 'vue-router'
import VueCompositionAPI from '@vue/composition-api'
import GetTextPlugin from 'vue-gettext'

import stubs from '@/tests/unit/stubs'

import SharesNavigation from '@files/src/components/AppBar/SharesNavigation'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(DesignSystem)
localVue.use(VueRouter)
localVue.use(VueCompositionAPI)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const router = new VueRouter()

describe('SharesNavigation component', () => {
  it('renders a shares navigation for both mobile and a desktop viewports', () => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
  })
})

function getWrapper() {
  return mount(SharesNavigation, {
    localVue,
    setup() {
      const navItems = [
        {
          to: '/files/shares/with-me/',
          text: 'Shared with me',
          active: false
        },
        {
          to: '/files/shares/with-others/',
          text: 'Shared with others',
          active: false
        },
        {
          to: '/files/shares/via-link/',
          text: 'Shared via link',
          active: false
        }
      ]
      const currentNavItem = navItems[0]
      return {
        navItems,
        currentNavItem
      }
    },
    router,
    stubs: {
      ...stubs,
      'oc-drop': false,
      'oc-list': false
    }
  })
}
