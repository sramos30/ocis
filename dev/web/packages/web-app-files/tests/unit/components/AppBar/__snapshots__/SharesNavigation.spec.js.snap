// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharesNavigation component renders a shares navigation for both mobile and a desktop viewports 1`] = `
<nav id="shares-navigation" aria-label="Shares pages navigation" class="oc-py-s">
  <ul class="oc-list oc-my-rm oc-mx-rm oc-flex oc-visible@s">
    <li>
      <oc-button-stub type="router-link" size="medium" to="/files/shares/with-me/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-mr-m oc-py-s shares-nav-desktop">
        <oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color=""></oc-icon-stub> <span>Shared with me</span>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="router-link" size="medium" to="/files/shares/with-others/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-mr-m oc-py-s shares-nav-desktop">
        <oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color=""></oc-icon-stub> <span>Shared with others</span>
      </oc-button-stub>
    </li>
    <li>
      <oc-button-stub type="router-link" size="medium" to="/files/shares/via-link/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-mr-m oc-py-s shares-nav-desktop">
        <oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="small" variation="passive" color=""></oc-icon-stub> <span>Shared via link</span>
      </oc-button-stub>
    </li>
  </ul>
  <div class="oc-hidden@s">
    <oc-button-stub type="button" size="medium" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" id="shares_navigation_mobile"><span>Shared with me</span>
      <oc-icon-stub name="arrow-down-s" filltype="line" accessiblelabel="" type="span" size="small" variation="passive" color=""></oc-icon-stub>
    </oc-button-stub>
    <div id="oc-drop-3" class="oc-drop oc-box-shadow-medium oc-rounded">
      <div class="oc-card oc-card-body oc-rounded oc-background-secondary oc-p-s">
        <ul class="oc-list oc-my-rm oc-mx-rm">
          <li>
            <oc-button-stub type="router-link" size="medium" to="/files/shares/with-me/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-my-xs shares-nav-mobile"><span class="icon-box"><oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub></span> <span>Shared with me</span></oc-button-stub>
          </li>
          <li>
            <oc-button-stub type="router-link" size="medium" to="/files/shares/with-others/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-my-xs shares-nav-mobile"><span class="icon-box"><oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub></span> <span>Shared with others</span></oc-button-stub>
          </li>
          <li>
            <oc-button-stub type="router-link" size="medium" to="/files/shares/via-link/" submit="button" variation="passive" appearance="raw" justifycontent="center" gapsize="medium" class="oc-my-xs shares-nav-mobile"><span class="icon-box"><oc-icon-stub name="info" filltype="fill" accessiblelabel="" type="span" size="medium" variation="passive" color=""></oc-icon-stub></span> <span>Shared via link</span></oc-button-stub>
          </li>
        </ul>
      </div>
    </div>
  </div>
</nav>
`;
